sdk/core/core-client/src/deserializationPolicy.ts

```diff
@@ -247,14 +243,26 @@ function handleErrorResponse(
           }
         }
         if (error.response) {
-          const errorResponse: FullOperationResponse = error.response;
-          errorResponse.parsedBody = operationSpec.serializer.deserialize(
+          deserializedError = operationSpec.serializer.deserialize(
             defaultBodyMapper,
             valueToDeserialize,
             "error.response.parsedBody"
```

Comment:
This is trying to deserialize `valueToDeserialize.error.response.parsedBody` ? Is there a reason we don't pass that property in as the `valueToDeserialize` instead?

Reply:
I was trying to keep the current code/logic. It deserializes and assigns to `error.response.parsedBody`  I was a little surprised that the code
```typescript
          const errorResponse: FullOperationResponse = error.response;
          errorResponse.parsedBody = operationSpec.serializer.deserialize(
```
also updates `error.response.parsedBody`

Reply:
There is some extra handling of sequence type I think that's why `valueToDeserialize` is introduced.

Reply:
This const is also introduced in last PR.  @sarangan12 do you remember the reason why this change?

Reply:
I guess it's to cast `error.response` back to `FullOperationResponse` so we can assign the `parsedBody` property.  I updated this PR to remove the if check on `error.response`, and do `as FulloperationResponse` cast.

Reply:
I feel like this could be cleaned up more, but I am too scared to suggest any specific changes this close to release, so I think I'm good with it.